### PR TITLE
1password-nightly (new cask)

### PR DESCRIPTION
**Pull request by duffpop**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask-versions/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by bevanjkay** [comment]
@duffpop We do not currently accept casks that contain `url do` blocks in this repo. 

Is there a link to download the `latest` version that we could use instead?

**Comment by duffpop** [comment]
> @duffpop We do not currently accept casks that contain `url do` blocks in this repo.
> 
> Is there a link to download the `latest` version that we could use instead?

Oh... then I'm confused @bevanjkay; the [CONTRIBUTING.md](https://github.com/Homebrew/homebrew-cask-versions/blob/master/CONTRIBUTING.md) explicitly states:

> the use of a [url do block](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#using-a-block-to-defer-code-execution) in conjunction with version :latest and sha256 :no_check is encouraged, to ensure the latest version is always fetched

Has the suggested procedure for nightly builds changed? I assume the Contributing doc is outdated as the [url do block](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#using-a-block-to-defer-code-execution) links to a 404.

Annoyingly, there isn't a static resource one can download `latest` from; 1Password maintains a changelog that the aforementioned Casks utilise. I'm happy to host my own tap if there isn't a graceful way to tackle this Cask, though I'd welcome any feedback you have.

File: Casks/1password-beta.rb [hunk 1]
```diff
@@ -12,11 +12,13 @@
...
     url "https://app-updates.agilebits.com/product_history/OPM#{version.major}"
-    regex(%r{href=.*?/1Password[._-]?v?(\d+(?:.\d+)*(?:[._-]BETA))[._-]?\$ARCH\.zip}i)
+    regex(
+      %r{href=.*?/1Password[._-]?v?(\d+(?:.\d+)*(?:[._-]BETA))[._-]?\$ARCH\.zip}i,
+    )
   end
 
   auto_updates true
-  conflicts_with cask: "1password"
+  conflicts_with cask: %w[1password homebrew/cask-versions/1password-nightly]
```
Comment (miccal) [inline Casks/1password-beta.rb:21]:
```suggestion
  conflicts_with cask: [
    "1password",
    "1password-nightly",
  ]
```

File: Casks/1password-nightly.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    next if latest_build_info.nil?
+
+    version = latest_build_info.match(/(\d+\.\d+\.\d+-\d+\.NIGHTLY)/)[0]
+    "https://downloads.1password.com/mac/1Password-#{version}-#{arch}.zip"
+  end
+  name "1Password Nightly"
+  desc "Password manager that keeps all passwords secure behind one password"
+  homepage "https://1password.com/"
+
+  conflicts_with cask: %w[1password homebrew/cask-versions/1password-beta]
```
Comment (miccal) [inline Casks/1password-nightly.rb:33]:
```suggestion
  conflicts_with cask: [
    "1password",
    "1password-beta",
  ]
```

File: Casks/1password-nightly.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          .first
+        next if match.nil?
+
+        match
+      end
+    next if latest_build_info.nil?
+
+    version = latest_build_info.match(/(\d+\.\d+\.\d+-\d+\.NIGHTLY)/)[0]
+    "https://downloads.1password.com/mac/1Password-#{version}-#{arch}.zip"
+  end
```
Comment (bevanjkay) [inline Casks/1password-nightly.rb:28]:
```suggestion
  url "https://c.1password.com/dist/1P/mac8/1Password-latest.NIGHTLY-#{arch}.zip"
```

**Review by bevanjkay — CHANGES_REQUESTED** [review]
Can you please re-add `auto_updates true`?

File: Casks/1password-beta.rb [hunk 4]
```diff
@@ -12,11 +12,13 @@
 
   livecheck do
     url "https://app-updates.agilebits.com/product_history/OPM#{version.major}"
-    regex(%r{href=.*?/1Password[._-]?v?(\d+(?:.\d+)*(?:[._-]BETA))[._-]?\$ARCH\.zip}i)
+    regex(
+      %r{href=.*?/1Password[._-]?v?(\d+(?:.\d+)*(?:[._-]BETA))[._-]?\$ARCH\.zip}i,
+    )
```
Comment (reitermarkus) [inline Casks/1password-beta.rb:17]: Leave this as a single line.

**Pull request closed without merging**